nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -258,6 +284,106 @@ contract StakingEscrow is Upgradeable, IERC900History {
         emit Deposited(_staker, _value);
     }
 
+    /**
+    * @notice Withdraw available amount of NU tokens to staker
+    * @param _value Amount of tokens to withdraw
+    */
+    function withdraw(uint256 _value) external onlyStaker {
+        require(_value > 0, "Value must be specified");
+        StakerInfo storage info = stakerInfo[msg.sender];
+        require(info.flags.bitSet(MERGED_INDEX), "Merge must be confirmed");
+        require(
+            _value + tStaking.stakedNu(info.operator) <= info.value,
+            "Not enough tokens unstaked in T staking contract"
+        );
+        require(
+            _value + getVestedTokens(msg.sender) <= info.value,
+            "Not enough tokens released during vesting"
+        );
+        info.value -= _value;
+
+        token.safeTransfer(msg.sender, _value);
+        emit Withdrawn(msg.sender, _value);
+    }
+
+    /**
+    * @notice Returns amount of not released yet tokens for staker
```

Comment:
This comment is a little confusing - makes it sound more like unvested as opposed to vested.

Is this analogous to what you intended?

```suggestion
    * @notice Returns tokens available for release to the staker based on vesting.
```

OR

```suggestion
    * @notice Returns tokens available to the staker based on vesting.
```

Unsure if there were already terminology discussions amongst yourselves?

Reply:
Terminology is not set and it looks like I did mistake during naming. Appreciate any help
So method doc is proper which means that this tokens are still locked. `getUnvestedTokens` ?

Reply:
From what I gather I think it might just be a terminology thing. 

So based on the line:

```python
uint256 vestedTokens = (info.vestingReleaseTimestamp - block.timestamp) * info.vestingReleaseRate;
        return info.value < vestedTokens ? info.value : vestedTokens;
```

Does 'release' mean the tokens have vested and are now available to be withdrawn? If so, there are still two parts to the calculation: 1) how many tokens have vested (`vestedTokens`) 2) how many tokens are currently staked `info.value`. 

So the value returned is really either the tokens vested, or the amount of tokens remaining because perhaps some tokens have already been withdrawn. Is that correct?

If the above is correct, the value returned is the tokens that can be withdrawn by the staker based on the vesting schedule - or am I missing something.

Reply:
Renamed to `unvested`